dev-db/influxdb/influxdb-2.7.1.ebuild

```diff
@@ -0,0 +1,238 @@
+	serde_derive-1.0.136
+	serde_json-1.0.79
+	serde_repr-0.1.7
+	smallvec-1.7.0
+	strsim-0.8.0
+	structopt-0.3.26
+	structopt-derive-0.4.18
+	syn-1.0.81
+	tempfile-3.3.0
+	termcolor-1.1.2
+	textwrap-0.11.0
+	thiserror-1.0.30
+	thiserror-impl-1.0.30
+	time-0.1.43
+	tinytemplate-1.2.1
+	tinyvec-1.5.1
+	tinyvec_macros-0.1.0
+	typed-arena-2.0.1
+	unicase-2.6.0
+	unicode-bidi-0.3.7
+	unicode-normalization-0.1.19
+	unicode-segmentation-1.8.0
+	unicode-width-0.1.9
+	unicode-xid-0.2.2
+	url-2.2.2
+	vcpkg-0.2.15
+	vec_map-0.8.2
+	version_check-0.9.3
+	walkdir-2.3.2
+	wasi-0.10.2+wasi-snapshot-preview1
+	wasm-bindgen-0.2.78
+	wasm-bindgen-backend-0.2.78
+	wasm-bindgen-macro-0.2.78
+	wasm-bindgen-macro-support-0.2.78
+	wasm-bindgen-shared-0.2.78
+	web-sys-0.3.55
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+"
+
+inherit cargo go-module
+
+DESCRIPTION="Scalable datastore for metrics, events, and real-time analytics"
+HOMEPAGE="https://www.influxdata.com"
+
+SRC_URI="https://github.com/influxdata/influxdb/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" https://github.com/influxdata/ui/releases/download/OSS-v${PV}/build.tar.gz -> ${P}-assets.tar.gz"
+SRC_URI+=" https://gentoo.kropotkin.rocks/go-pkgs/${P}-deps.tar.xz"
+SRC_URI+=" $(cargo_crate_uris)"
+
+LICENSE="Apache-2.0 BSD BSD-2 EPL-2.0 ISC MIT MPL-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="man test"
+RESTRICT="!test? ( test )"
+
+BDEPEND="man? (
+	app-text/asciidoc
+	app-text/xmlto
+)"
+
+COMMON_DEPEND="
+	acct-group/influxdb
+	acct-user/influxdb
+"
+
+DEPEND="${COMMON_DEPEND}"
+
+RDEPEND="${COMMON_DEPEND}"
+
+src_unpack() {
+	cargo_src_unpack || die
+}
+
+src_prepare() {
+	default
+
+	DATA_DIR="${S}/static/data"
+	mkdir "${DATA_DIR}" || die
+	mv "${WORKDIR}/build"  "${DATA_DIR}" || die
+}
+
+src_compile() {
+	(
+		mv "${WORKDIR}/go-mod" "${WORKDIR}/go-mod-tmp" || die
+		mv "${WORKDIR}/go-mod-tmp/github.com/influxdata/pkg-config@v0.2.11/go-mod" "${WORKDIR}/go-mod" || die
+		cd ${WORKDIR}/go-mod-tmp/github.com/influxdata/pkg-config* || die
+		ego build . || die
+		mv "${WORKDIR}/go-mod" "${WORKDIR}/go-mod-tmp/github.com/influxdata/pkg-config@v0.2.11"
+		mv "${WORKDIR}/go-mod-tmp" "${WORKDIR}/go-mod"
+	)
+
+	(
+		cd "${WORKDIR}/go-mod/github.com/influxdata/flux@v0.193.0/libflux" || die
+		cargo_src_compile || die
+	)
+
+	export PKG_CONFIG="${WORKDIR}/go-mod/github.com/influxdata/pkg-config@v0.2.11/pkg-config"
+	go generate ./static || die
+	GOBIN="${S}/bin" \
+		go install \
+			-tags 'assets,noasm,sqlite_json,sqlite_foreign_keys' \
+			-ldflags="-X main.version=${PV}" \
+		./... || die "compile failed"
+	use man && emake -C man build
+}
+
+src_install() {
+	dobin bin/influx*
+	dodoc *.md
+	use man && doman man/*.1
+
+	newconfd "${FILESDIR}"/influxdb.confd influxdb
+	newinitd "${FILESDIR}"/influxdb.initd influxdb
+	keepdir /var/log/influxdb
+	fowners influxdb:influxdb /var/log/influxdb
+}
+
+src_test() {
+	ego test ./tests || die
+}
```

Comment:
Testing is done between `src_compile` and `src_install` so move the phase to its right place please.